### PR TITLE
Consistently apply order across components

### PR DESCRIPTION
**Pull request by davkean**
Our components had a different precendence based on whomever wrote the code. This applies a precendence across the entire tree that is consistent and allows components to predictably run before or after us.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        /// <summary>
+        ///     Represents the default order precedence for managed components.
+        /// </summary>
+        public const int Default = 10;
+
+        /// <summary>
+        ///     Represents the an order precedence for managed components that is ordered before <see cref="Default"/>.
+        /// </summary>
+        public const int BeforeDefault = 20;
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs]: Completely open for feedback on these names - I spent a little bit time trying to come up with better names but failed. If you have some suggestions that would be great!

File: src/Microsoft.VisualStudio.ProjectSystem.FSharp.VS/ProjectSystem/VS/Imaging/FSharpProjectImageProvider.cs [hunk 2]
```diff
@@ -11,7 +11,6 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Imaging
     /// </summary>
     [Export(typeof(IProjectImageProvider))]
     [AppliesTo(ProjectCapability.FSharp)]
-    [Order(2)]
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.FSharp.VS/ProjectSystem/VS/Imaging/FSharpProjectImageProvider.cs]: @srivatsn I couldn't see how this was applied - it appears deliberate. Why does it need it? I can't expect more than one of these in place at a time.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        /// <summary>
+        ///     Represents the lowest order precedence for components.
+        /// </summary>
+        public const int Lowest = 0;
+
+        /// <summary>
+        ///     Represents the default order precedence for managed components.
+        /// </summary>
+        public const int Default = 10;
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs]: Note, this value is deliberate as @BillHiebert makes use of it in their overrides of the launch provider stuff.

File: src/Microsoft.VisualStudio.ProjectSystem.FSharp.VS/ProjectSystem/VS/Imaging/FSharpProjectImageProvider.cs (hunk 2 above)
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.FSharp.VS/ProjectSystem/VS/Imaging/FSharpProjectImageProvider.cs]: This is not needed anymore. Originally I had added the CSharp capability to F# projects as a quick way of getting everything to work and in that case, there were two providers providing icons and needed to be ordered. But now that I've removed that hack, there'll be just one and so we don't need this anymore.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs (hunk 3 above)
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs]: `Default` immediately meant to me as `0` which is CPS's default. I don't know how to represent, this `Default` is `ManagedDefault` and not the system wide default.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs (hunk 1 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs]: @dotnet/project-system Come on folks, I need better names than this. :)

Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs]: If you're still waiting for a better name I don't think you're going to get one. What's wrong with BeforeDefault? It's before the default rules - great name - go ahead :)

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Order.cs]: Yeah, not really happy with the names...

**Pull request merged**